src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -49,19 +50,23 @@ public abstract class DetailControlModel : INotifyPropertyChanged, IDisposable
 
         private Dictionary<NuGetVersion, DetailedPackageMetadata> _metadataDict = new Dictionary<NuGetVersion, DetailedPackageMetadata>();
 
-        private INuGetUI _uiController;
+        private readonly ISettings _settings;
+        private readonly INuGetUI _uiController;
 
         protected DetailControlModel(
             IServiceBroker serviceBroker,
             IEnumerable<IProjectContextInfo> projects,
-            INuGetUI uiController)
+            INuGetUI uiController,
+            ISettings settings)
         {
             _nugetProjects = projects;
             ServiceBroker = serviceBroker;
             _uiController = uiController;
+            _settings = settings ?? throw new ArgumentNullException(nameof(settings));
+            _settings.SettingsChanged += Settings_SettingsChanged;
```

Comment:
events must always be unsubscribed, or there will be memory leaks.

The event (`_settings.SettingsChanged`) will hold a reference to this instance of `DetailControlModel`, and therefore it will not be garbage collected until `_settings` is GC'd.  However, `VsSettings` is a singleton, and therefore none of the `DetailControlModels` will be GC'd during the process lifetime. For this reason, the event unsubscribe cannot happen in a `Dispose()` method, unless something explicitly calls dispose on the instance, since the automatic dispose on GC will never happen.